repos/system_upgrade/el7toel8/actors/selinux/selinuxprepare/tests/component_test.py

```diff
@@ -7,95 +7,86 @@
 from leapp.libraries.stdlib import api, run, CalledProcessError
 from leapp.reporting import Report
 
-enabled = False
 
-test_modules = [
+TEST_MODULES = [
     ["400", "mock1"],
     ["99", "mock1"],
     ["300", "mock1"],
     ["400", "mock2"],
     ["999", "mock3"]
 ]
 
-semanage_commands = [
+SEMANAGE_COMMANDS = [
     ['fcontext', '-t', 'httpd_sys_content_t', '"/web(/.*)?"'],
     ['fcontext', '-t', 'ganesha_var_run_t', '"/ganesha(/.*)?"'],
     ['port', '-t', 'http_port_t', '-p', 'udp', '81'],
     ['permissive', 'abrt_t']
 ]
 
-# save value of semodule -lfull for comparison
-semodule_lfull = ""
-semanage_export = ""
+testmoduledir = "tests/mock_modules/"
 
-if enabled:
+
+def _run_cmd(cmd, split=False, logmsg=""):
     try:
-        semodule = run(["semodule", "-lfull"], split=False)
-        semodule_lfull = semodule.get("stdout", "")
-        semanage = run(["semanage", "export"], split=False)
-        semanage_export = semanage.get("stdout", "")
+        return run(cmd, split=split).get("stdout", "")
     except CalledProcessError as e:
-        api.current_logger().warning("Error listing SELinux customizations: %s", str(e.stderr))
+        # XXX FIXME not sure logging in tests makes sense
+        api.current_logger().warning("%s: %s", logmsg, str(e.stderr))
+
 
-testmoduledir = os.path.join(os.getcwd(), "tests/mock_modules/")
+SEMODULE_LFULL_INITIAL = _run_cmd(["semodule", "-lfull"], logmsg="Error listing SELinux customizations")
+SEMANAGE_EXPORT_INITIAL = _run_cmd(["semanage", "export"], logmsg="Error listing SELinux customizations")
 
 
 def setup():
-    if enabled:
-        for priority, module in test_modules:
-            try:
-                run(["semodule", "-X", priority, "-i", os.path.join(testmoduledir, module + ".cil")])
-            except CalledProcessError as e:
-                api.current_logger().warning("Error installing mock module: %s", str(e.stderr))
-                continue
-
-        for command in semanage_commands:
-            try:
-                run(["semanage", command[0], "-a"] + command[1:])
-            except CalledProcessError as e:
-                api.current_logger().warning("Error applying selinux customizations %s", str(e.stderr))
-                continue
-
-
-@pytest.mark.skip(reason='Test disabled because it would modify the system')
+    # NOTE(ivasilev) Maybe there is a more elegant way to conditionally run setup/teardown
+    enabled = os.environ.get("DESTRUCTIVE_TESTING", False)
```

Comment:
well besides it probably is a bad idea just to check for the existence of an environment variable. If someone sets it to '0', I wouldn't expect this to trigger anyway but here it would.

Reply:
Fixed